File: DarkOrbit 10.0/Net/netty/commands/PetInitializationCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class PetInitializationCommand
    {
        public const short ID = 9174;

        public static byte[] write(bool hasPet, bool hasFuel, bool petIsAlive)
        {
            var param1 = new ByteArray(ID);
            param1.writeBoolean(hasPet);
            param1.writeBoolean(hasFuel);
            param1.writeBoolean(petIsAlive);
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/PetInitializationCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

/**
 Created by Pedro on 17-03-2015.
 */
public class PetInitializationCommand
        implements ServerCommand {

    public static int ID = 32119;

    public boolean hasPet;
    public boolean hasFuel;
    public boolean petIsAlive;

    public PetInitializationCommand(boolean param1, boolean param2, boolean param3) {
        this.hasFuel = param1;
        this.petIsAlive = param2;
        this.hasPet = param3;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeBoolean(this.hasPet);
            param1.writeBoolean(this.hasFuel);
            param1.writeShort(28461);
            param1.writeBoolean(this.petIsAlive);
            param1.writeShort(-21872);
        } catch (IOException e) {
        }
    }

}
